web/src/main/java/se/inera/intyg/webcert/web/service/facade/list/ResourceLinkListHelperImpl.java

```diff
@@ -174,8 +176,12 @@ private ResourceLinkDTO getConvertedResourceLink(
     }
 
     private boolean validateRenew(ActionLink link, String certificateType, CertificateRelations relations, CertificateStatus status) {
-        return link.getType() == ActionLinkType.FORNYA_INTYG
-            && CertificateRenewFunction.validate(certificateType, relations, status, authoritiesHelper);
+        final var actionLinkIsRenewCertificate = link.getType() == ActionLinkType.FORNYA_INTYG;
```

Comment:
Ah, tricky that we have to handle the situation when we have a correct link and when we have to validate it with CertificateRenewFunction.validate(...).

Now we will call cs everytime we need to check activeAndSupportsType. This will be costly as it will be done in a loop and  for each certificate. Could we "rearrange" things to remove the need of calling cs?